scripts/migrate-to-testnet.ts

```diff
@@ -0,0 +1,255 @@
+    fs.mkdirSync(WASM_DIR, { recursive: true });
+}
+
+async function downloadWasm(codeId: number, contractName: string): Promise<Uint8Array> {
+    const wasmPath = path.join(WASM_DIR, `${contractName}_${codeId}.wasm`);
+
+    // Check if we already have the file
+    if (fs.existsSync(wasmPath)) {
+        console.log(`Using cached WASM file for ${contractName} (${codeId})`);
+        return new Uint8Array(fs.readFileSync(wasmPath));
+    }
+
+    try {
+        const response = await axios.get(
+            `${MAINNET_API}/cosmwasm/wasm/v1/code/${codeId}`,
+            { 
+                headers: {
+                    'Accept': 'application/json'
+                }
+            }
+        );
+        
+        console.log(`Retrieved code info for ${contractName} (ID: ${codeId})`);
+        
+        // The data field contains the base64-encoded WASM
+        const base64Wasm = response.data.data;
+        if (!base64Wasm) {
+            throw new Error(`No WASM data found for code ID ${codeId}`);
+        }
+
+        // Convert base64 to Uint8Array
+        const binaryString = Buffer.from(base64Wasm, 'base64');
+        const wasmBinary = new Uint8Array(binaryString);
+
+        // Save to file
+        fs.writeFileSync(wasmPath, wasmBinary);
+        console.log(`Saved WASM file to ${wasmPath}`);
+
+        return wasmBinary;
+
+    } catch (error) {
+        console.error('Error downloading WASM:', error.response?.data || error.message);
+        throw error;
+    }
+}
+
+async function main() {
+    // Read contracts.json
+    const contractsData: ContractInfo[] = JSON.parse(fs.readFileSync(CONTRACTS_FILE, 'utf8'));
+    
+    // Initialize or read existing testnet contracts
+    let testnetContracts: TestnetContractInfo[] = [];
+    if (fs.existsSync(TESTNET_CONTRACTS_FILE)) {
+        testnetContracts = JSON.parse(fs.readFileSync(TESTNET_CONTRACTS_FILE, 'utf8'));
+    }
+    
+    // Process all contracts
+    const contractsToProcess = contractsData;
+    console.log(`Processing all contracts: ${contractsToProcess.map(c => c.name).join(', ')}`);
+
+    // Get wallet from mnemonic
+    const mnemonic = process.env.MNEMONIC;
+    if (!mnemonic) {
+        throw new Error("Please set MNEMONIC environment variable");
+    }
+
+    const wallet = await DirectSecp256k1HdWallet.fromMnemonic(mnemonic, {
+        prefix: "xion",
+    });
+    const [account] = await wallet.getAccounts();
+
+    // Connect to mainnet to get code IDs
+    const mainnetClient = await SigningCosmWasmClient.connect(MAINNET_RPC);
+    
+    // Connect to testnet for deployment with gas price
+    const gasPrice = GasPrice.fromString('0.025uxion');
+    const testnetClient = await SigningCosmWasmClient.connectWithSigner(
+        TESTNET_RPC,
+        wallet,
+        { gasPrice }
+    );
+
+    console.log(`Starting migration with address: ${account.address}`);
+
+    // Process each contract
+    for (const contract of contractsToProcess) {
```

Comment:
Right now this is going to download the wasm for every contract every time its ran. Can we limit this to the contracts that haven't been uploaded to testnet?